### DISCUSSION TITLE
Generate csv of failures

### DISCUSSION THREAD

**Discussion by phirestalker** [jsaddiction/TrailerTech#52] [Ideas]
My movie collection is more than 500. Anticipating many failures (since youtube is a source), I set my logging to warning hoping to pick up only the failed downloads. Looking through the log file I noticed that when set to warning I only get the error and sometimes not the movie it is about.

I have over 200 trailers that could not be downloaded, so sifting through the log file is not really practical. I thought a file containing only final errors (trailers that had no successful downloads at all) would be useful.

Would it be possible to have a CSV file created that would contain title, year, TMDb, IMDb, failure reason (missing ids, bad trailer URL, etc)? This way I could easily sort by error and could knock out batches at a time (say fixing TMDb trailer URLs).

Great script by the way. I wish radarr did this internally, but this script makes everything very easy. Even though you wrote it for Plex, and Kodi, the files are picked up by Jellyfin as well.

**Comment by jsaddiction** [comment]
I use jellyfin and kodi with this script and for me it's working pretty well. I have just over 1400 titles and 2 missing trailers. I suspect that most of your errors are a result of the script being unable to determine the correct movie.

For instance, it first looks in a directory for an nfo file that contains details of the movie. Specially it looks for imdb, tmdb, year, and title. At a minimum it needs one of the id's or the title and year. If none of that can be parsed, it looks to the directory name and parses "title (year)". In addition it will try parsing a movies file name "title (imdb)". If none of that is found then it gives up.

If your dataset meets those conditions then there may be some other issues going on like what you mentioned "tmdb doesn't have a record" in that case your right it will take a while to populate those trailers into the db

Thank you for the compliment! I would like to develop this into a web app similar to the *arr software but at the moment, the task seems pretty daunting.

**Reply by phirestalker** [reply]
I just remembered that I also wanted to mention that since logging is done as it happens, I have no way of knowing if a trailer was ultimately found.
I just looked at the log file, and there is only one error about not finding the movie. That was for a hidden directory so it is irrelevant. It looks like all the errors are about too small or failed to download. I noticed some failed to remove and failed to move, which I will have to fix some permissions for.
I am willing to take a stab at coding the CSV bit, I just wanted to put the idea on your radarr (pun intended). I see that there is a status variable as to whether a trailer was ultimately downloaded. Are the errors collected somewhere as well, or will I need to implement that?

**Reply by jsaddiction** [reply]
The errors are not collected. This script is very linear in which once a directory has been processed, error or not, it continues to the next.

You could call it by the cli for a specific directory in debug mode to help with troubleshooting.

Most trailers, around 90%, come from YouTube so if that is disabled in your config then that could be another reason you aren't getting anything.

Try a directory by itself and adjust the config until it gathers the trailers you like, then run it on your full dataset.

Adding failures to a csv file is only really helpful during troubleshooting and operations from the cli. When radarr calls it, all info is present in env variables.

You are welcome to fork the repo and modify anyway you like but at this stage, I'm not really motivated to increase it's complexity with a merge. It really needs a rewrite with a database backend and a web gui front-end

**Reply by phirestalker** [reply]
aha! You just gave me an idea. I can run some fancy bash-fu to list directories that do not contain files that end with -preview.*, then I will have the failed movies.
